Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugAdapterLauncher.cs

```diff
@@ -0,0 +1,213 @@
+            for (int i = 0; i < options.Length; i++) {
+                if (options[i] == ';') {
+                    if (i < options.Length - 1 && options[i + 1] != ';') {
+                        // valid option boundary
+                        res.Add(options.Substring(lastStart, i - lastStart));
+                        lastStart = i + 1;
+                    } else {
+                        i++;
+                    }
+                }
+            }
+            if (options.Length - lastStart > 0) {
+                res.Add(options.Substring(lastStart, options.Length - lastStart));
+            }
+            return res.ToArray();
+        }
+
+        #endregion
+
+        #region Attach
+        public static DebugAttachInfo GetTcpAttachDebugInfo(IAdapterLaunchInfo adapterLaunchInfo) {
+            DebugAttachInfo debugAttachInfo = new DebugAttachInfo();
+
+            adapterLaunchInfo.DebugPort.GetPortName(out string adapterHostPortInfo);
+            debugAttachInfo.RemoteUri = new Uri(adapterHostPortInfo);
+
+            var uriInfo = new Uri(adapterHostPortInfo);
+            debugAttachInfo.Host = uriInfo.Host;
+            debugAttachInfo.Port = uriInfo.Port;
+
+            return debugAttachInfo;
+        }
+
+        #endregion
+
+        private static void AddDebugOptions(IAdapterLaunchInfo adapterLaunchInfo, DebugInfo launchJson) {
+            var debugService = (IPythonDebugOptionsService)Package.GetGlobalService(typeof(IPythonDebugOptionsService));
+
+            // Stop on entry should always be true for VS Debug Adapter Host.
+            // If stop on entry is disabled then VS will automatically issue
+            // continue when it sees "stopped" event with "reason=entry".
+            launchJson.StopOnEntry = true;
+
+            launchJson.PromptBeforeRunningWithBuildError = debugService.PromptBeforeRunningWithBuildError;
+            launchJson.RedirectOutput = debugService.TeeStandardOutput;
+            launchJson.WaitOnAbnormalExit = debugService.WaitOnAbnormalExit;
+            launchJson.WaitOnNormalExit = debugService.WaitOnNormalExit;
+            launchJson.BreakOnSystemExitZero = debugService.BreakOnSystemExitZero;
+            launchJson.DebugStdLib = debugService.DebugStdLib;
+            launchJson.ShowReturnValue = debugService.ShowFunctionReturnValue;
+        }
+
+        [DllImport("shell32.dll")]
+        static extern IntPtr CommandLineToArgvW([MarshalAs(UnmanagedType.LPWStr)] string lpCmdLine, out int pNumArgs);
+
+        private static IEnumerable<string> GetParsedCommandLineArguments(string command) {
+            if (String.IsNullOrEmpty(command)) {
+                yield break;
+            }
+
+            IntPtr argPointer = CommandLineToArgvW(command, out int argumentCount);
+            string[] arguments = new string[argumentCount];
+
+            if (argPointer != IntPtr.Zero) {
+                try {
+                    for (int i = 0; i < arguments.Length; i++) {
```

Comment:
change `arguments.Length` to `argumentCount` and get rid of the otherwise unused `arguments`

Reply:
Will be updated in next commit